### PR TITLE
feat(condo): DOMA-8003 add acquiring fields to Invoice

### PR DESCRIPTION
**Pull request by VKislov**

### CODE REVIEW

File: apps/condo/domains/marketplace/schema/Invoice.js [hunk 1]
```diff
@@ -218,6 +220,44 @@ const Invoice = new GQLListSchema('Invoice', {
             type: 'DateTimeUtc',
         },
 
+        acquiringIntegrationId: {
+            schemaDoc: 'Type of user who changed the pass. Can be employee or resident',
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/marketplace/schema/Invoice.js:224]: update schemaDoc of new fields

File: apps/condo/domains/marketplace/schema/Invoice.js [hunk 2]
```diff
@@ -218,6 +220,44 @@ const Invoice = new GQLListSchema('Invoice', {
             type: 'DateTimeUtc',
         },
 
+        acquiringIntegrationId: {
+            schemaDoc: 'Integration ID through which this invoice can be paid',
+            type: 'Virtual',
+            resolver: async (item, args, context) => {
+                const invoiceContext = await getById('InvoiceContext', item.context)
```
Comment (AleX83Xpert) [inline apps/condo/domains/marketplace/schema/Invoice.js:227]: The `InvoiceContext` model is removed. Please do not use it. See #4142.
It's better to add only one virtual field that emulates context:
```js
context { integration { hostUrl canGroupReceipts currencyCode } }
```
To find these fields you should find the `AcquiringIntegrationContext` for the organization with finished `invoiceStatus`.

File: apps/condo/migrations/20231214184616-0353_invoicehistoryrecord_acquiringintegrationid_and_more.js
Comment (AleX83Xpert) [inline apps/condo/migrations/20231214184616-0353_invoicehistoryrecord_acquiringintegrationid_and_more.js:1]: Migration for virtual fields? Is it really necessary?

**Review by AleX83Xpert — CHANGES_REQUESTED** [review]

File: apps/condo/migrations/20231214184616-0353_invoicehistoryrecord_acquiringintegrationid_and_more.js
Comment (VKislov) [reply apps/condo/migrations/20231214184616-0353_invoicehistoryrecord_acquiringintegrationid_and_more.js:1]: ?????

Comment (AleX83Xpert) [reply apps/condo/migrations/20231214184616-0353_invoicehistoryrecord_acquiringintegrationid_and_more.js:1]: Hmmm, looks like it's normal. Sorry :)

Comment (VKislov) [reply apps/condo/migrations/20231214184616-0353_invoicehistoryrecord_acquiringintegrationid_and_more.js:1]: I tried using a field type declaration not as a string but as an object from `@keystonejs/fields`
the result is the same, the field is still included in the migration.

Comment (VKislov) [reply apps/condo/migrations/20231214184616-0353_invoicehistoryrecord_acquiringintegrationid_and_more.js:1]: I also did a check through a direct request to the database. When selecting from the database, instances arrive without these fields. Apparently some kind of migrator logic is special, I think you can find out something about it from @pahaz

File: apps/condo/migrations/20231219124252-0354_billingreceipthistoryrecord_acquiringintegrationid_and_more.js
Comment (nomerdvadcatpyat) [inline apps/condo/migrations/20231219124252-0354_billingreceipthistoryrecord_acquiringintegrationid_and_more.js:1]: need to update to 0355

**Review by nomerdvadcatpyat — APPROVED** [review]

File: apps/condo/domains/billing/schema/BillingReceipt.js [hunk 3]
```diff
@@ -209,6 +210,62 @@ const BillingReceipt = new GQLListSchema('BillingReceipt', {
...
+            schemaDoc: 'Can multiple receipts be united through this acquiring',
+            type: 'Virtual',
+            graphQLReturnType: 'Boolean',
+            resolver: async (item, args, context) => {
+                const billingContext = await getById('BillingIntegrationOrganizationContext', item.context)
+                const acquiringContext = await find('AcquiringIntegrationContext', { organization: { id: billingContext.organization }, status: CONTEXT_FINISHED_STATUS, deletedAt: null })
+                if (!isEmpty(acquiringContext)) {
+                    const acquiringIntegration = await getById('AcquiringIntegration', get(acquiringContext[0], 'integration', null))
+                    return get(acquiringIntegration, 'canGroupReceipts', null)
+                }
```
Comment (AleX83Xpert) [inline apps/condo/domains/billing/schema/BillingReceipt.js:254]: What about moving the fetching of AcquiringIntegration to a separate function?

File: apps/condo/domains/billing/schema/BillingReceipt.js [hunk 4]
```diff
@@ -209,6 +210,62 @@ const BillingReceipt = new GQLListSchema('BillingReceipt', {
...
+                const acquiringContext = await find('AcquiringIntegrationContext', { organization: { id: billingContext.organization }, status: CONTEXT_FINISHED_STATUS })
+                if (!isEmpty(acquiringContext)) {
+                    const acquiringIntegration = await getById('AcquiringIntegration', get(acquiringContext[0], 'integration', null))
+                    return get(acquiringIntegration, 'id', null)
+                }
+                return null
+            },
+        },
+
+        hostUrl: {
```
Comment (AleX83Xpert) [inline apps/condo/domains/billing/schema/BillingReceipt.js:229]: Just a question. Is this name ok? Maybe use `acquiringHostUrl`?

**Review by AleX83Xpert — APPROVED** [review]

**Pull request merged**